scalableminds/webknossos
# Accept terms of Service at signup

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

Accept terms of service already at sign up / organization creation.

Backend is done. Frontend needs to call /api/termsOfService route and add a checkbox / iframe to the account creation page. Then set "acceptedTermsOfService" in the submitted form to /api/auth/createOrganizationWithAdmin  to the TOS VERSION. The Terms of Service acceptance afterward can be removed.

### Steps to test:
- Register as a new user
- Try not to accept the Terms of Service, this should not be possible
- Switch `isWkOrgInstance` in application.conf, it should behave the same regarding the ToS
- Disable the ToS in application.conf, there should no longer be a ToS checkbox
- After having signed up, increase the ToS version in application.conf. The user should now see the ToS modal and have to accept the ToS

### TODOs:
     - [x] wkorg
     - [x] generic
- [x] Remove dev-only changes to application.conf

### Issues:
- fixes #8149 

------
(Please delete unneeded items, merge only when none are left open)
